test/integration/version_upgrade_test.go

```diff
@@ -35,6 +35,113 @@ import (
 	pkgutil "k8s.io/minikube/pkg/util"
 )
 
+// TestDifferentVersionUpgrade test installation of different version of Kubernetes.
+// Start with installing from v1.13.0 then to v1.15.0 and then try to downgrade it
+// to v1.13.0
+// Following are the test cases that the function is doing
+// 1. Start with specific version
+//    ./out/minikube start -p vupgrade --kubernetes-version=v1.13.0
+// 2. Upgrade with newer version
+//   ./out/minikube start -p vupgrade --kubernetes-version=v1.13.0
+// 3. Downgrade with older version (this will spit out error and test the error message)
+//   ./out/minikube start -p vupgrade --kubernetes-version=v1.13.0
+func TestDifferentVersionUpgrade(t *testing.T) {
```

Comment:
In the interest of keeping integration tests fast, I think this function can be safely removed as it does not add much over  `TestVersionUpgrade`.

If you really want to assert that we don't break the `downgrades are always failures` case, I would suggest adding it to `TestVersionUpgrade`, where it won't incur the penalty of creating an additional cluster.

Reply:
No worries. Will refactor the code. Thanks

Reply:
> In the interest of keeping integration tests fast, I think this function can be safely removed as it does not add much over `TestVersionUpgrade`.
> 
> If you really want to assert that we don't break the `downgrades are always failures` case, I would suggest adding it to `TestVersionUpgrade`, where it won't incur the penalty of creating an additional cluster.

Gone through the `TestVersionUpgrade(..) `logic and even though in terms of what it is trying to accomplish (checking kubernetes version updrade checking) is similar but the checking that it is performing is less than what `TestDifferentVersionUpgrade(..) `is doing. The `TestDifferentVersionUpgrade(..)` checks the output to make sure that the process does execute properly while `TestVersionUpgrade(..) `does not do any kind of checking as it just 'assume' that the process will start.

My thought behind this is that by having test cases that checks the output will assure that there are no behavioural changes in the code. 

If this is not valid then happy to remove to just integrate the test case (3) for checking the downgrade to lower version to `TestVersionUpgrade(..)`

Reply:
Rather than implementing the test twice, let's improve `TestVersionUpgrade` so that both tests benefit.  The only thing missing is:

- A version check (see my later comments)
- A downgrade check

The downgrade check can just come at the end of the test. No need to assert the output, just check that start fails if you try to downgrade to the same version that the test started with.

Reply:
I've updated refactor the code as per the suggested comment. `TestVersionUpgrade` now contains 2 additional test cases.